crates/ruma-federation-api/src/membership/create_invite/v2.rs

```diff
@@ -39,6 +41,14 @@ pub struct Request {
 
     /// An optional list of simplified events to help the receiver of the invite identify the room.
     pub invite_room_state: Vec<Raw<AnyStrippedStateEvent>>,
+
+    /// An optional list of servers the invited homeserver should attempt to join or leave via, as
+    /// according to [MSC4125](https://github.com/matrix-org/matrix-spec-proposals/pull/4125).
+    ///
+    /// The future key for this if accepted will be `via`
+    #[cfg(feature = "unstable-msc4125")]
+    #[serde(default, rename = "org.matrix.msc4125.via")]
+    pub via: Option<Vec<OwnedServerName>>,
```

Comment:
Since the behavior should be the same whether the `Vec` is empty or absent, I believe we don't need to wrap it in an `Option`. We should use `skip_serializing_if = "Vec::is_empty"` though.

Reply:
[An empty `Vec` is explicitly forbidden](https://github.com/matrix-org/matrix-spec-proposals/pull/4125/files#diff-b2b3a2f3876ea102465cd605597449ad4668ecf060eda10d6bb9d3fa5faf3c0fR65-R66)

Reply:
It's not a problem, since we don't serialize it.

Reply:
Ah but then you can't enforce the validation.

Reply:
Oh, it seems I misunderstood what `skip_serializing_if` does, but now I realized it's actually very obvious.